src/qrules/io/_dot.py

```diff
@@ -513,7 +514,7 @@ def _collapse_graphs(
             FrozenTransition(
                 topology,
                 states={
-                    i: tuple(sorted(particles, key=lambda p: p.name))
+                    i: tuple(sorted(particles))  # ??? make more generic sorting?
```

Comment:
Same here: you can extract a function that you can then provide to `sorted(..., key=function)`.

```python
def _sorting_key(obj: Any) -> Hashable:
    if isintance(obj, Particle):
        return obj.name
    ...
    if isintance(obj, abc.Hashable):
        return obj
    raise NotImplementedError
```